src/dataStructures/linkedList/LinkedList.java

```diff
@@ -245,13 +253,13 @@ public void sort() {
     int mid = (this.size - 1) / 2;
     Node<T> middle = this.get(mid);
     Node<T> nextHead = middle.next;
-    LinkedList<T> next = new LinkedList<>(nextHead, this.size - 1 - mid);
+    LinkedList<T> next = new LinkedList<>(nextHead, this.size - 1 - mid); // Split the list into 2.
     middle.next = null;
-    this.size = mid + 1; // update size of original LL after split
+    this.size = mid + 1; // update size of original list after split
 
-    next.sort();
+    next.sort(); // Recursively split the list.
     this.sort();
-    head = merge(this, next);
+    head = merge(this, next); // Combine the split lists.
```

Comment:
technically it isn't any combine. Its merging the two sorted lists to form an entire sorted linked list. But this is rather verbose to say it out so omitting the explanation here is fine too. Merge routine is understood to be a necessity when doing merge sort.